app/src/test/java/ch/hikemate/app/model/route/HikesViewModelTest.kt

```diff
@@ -878,6 +878,42 @@ class HikesViewModelTest {
         assertNull(hikesViewModel.selectedHike.value)
       }
 
+  @Test
+  fun `unsaveHike also unplans hike`() =
+      runTest(dispatcher) {
+        // Make sure the loaded hike is included in the saved hikes
+        coEvery { savedHikesRepo.loadSavedHikes() } returns
+            listOf(
+                SavedHike(
+                    id = singleOsmHike1[0].id,
+                    name = singleOsmHike1[0].name ?: "",
+                    date = firstJanuary2024))
+        hikesViewModel.refreshSavedHikesCache()
```

Comment:
Nitpick: you can make use of the util `loadSavedHikes` here. The subsequent `loadOsmHikes(singleOsmHike1)` call you make will override the loaded hikes anyway, so it does not matter here whether you use refresh or load saved hikes.

```suggestion
        loadSavedHikes(listOf(
            SavedHike(
                    id = singleOsmHike1[0].id,
                    name = singleOsmHike1[0].name ?: "",
                    date = firstJanuary2024)))
```